### DISCUSSION TITLE
current_user isn't correct type for adding SQLAlchemy relations

### DISCUSSION THREAD

**Discussion by mdtomo** [fastapi-users/fastapi-users#918] [Q&A] [answered]
I'm currently trying to create a new record and add the current_user to the related user field. The `current_user` isn't an actual database instance of the user. Instead it's a Pydantic type (UserDB).

The only way I can get this to work is by creating another query for the user.

Is this expected behaviour or is `Depends(deps.get_current_active_user)` returning the wrong type?

Example from https://github.com/rafsaf/minimal-fastapi-postgres-template

```
@router.post("/", response_model=Project, status_code=201)
async def create_project(
    create_project: CreateProject,
    session: AsyncSession = Depends(deps.get_session),
    current_user: models.UserTable = Depends(deps.get_current_active_user),
) -> Any:
    """
    Creates a new project
    """
    new_project = models.Project(name=create_project.name)
    user_query = select(models.UserTable).where(models.UserTable.id == current_user.id)
    result = await session.execute(user_query)
    user = result.unique().scalars().first()

    new_project.users.append(user) # current_user here will raise AttributeError: 'UserDB' object has no attribute '_sa_instance_state'
    
    session.add(new_project)
    await session.commit()
    await session.refresh(new_project)
    return new_project
```

**Comment by frankie567** [comment, marked as answer]
Yes, it's expected. Since we need to be database/ORM-agnostic, the user is always returned as a Pydantic model.

An alternative way to do what you show in the example is to directly create association object instead of appending to the users list on `project`.
